lib/rspec/core/filter_manager.rb

```diff
@@ -46,7 +46,7 @@ def prune(examples)
 
           examples.select do |ex|
             file_scoped_include?(ex.metadata, ids, locations) do
-              !exclusions.include_example?(ex) && non_scoped_inclusions.include_example?(ex)
+              (exclusions.empty? || !exclusions.include_example?(ex)) && non_scoped_inclusions.include_example?(ex)
```

Comment:
Why? (not had enough ☕  yet 😂 )

Reply:
This was *the* hardest part of the change.
Without this, `exclusions.include_example?(ex)` would just always return `true`, and its negation - `false`.
This is due to the change in the underlying logic in the case when there are no rules, that is coming from this:
```ruby
[].all?(&:odd) # new version => true
[].any?(&:odd) # before => false
```

Without this check, `rspec` always finishes with `0 examples, 0 failures`.

Reply:
`InclusionFilter` has a similar check:
```ruby
      def include_example?(example)
        @rules.empty? || super
      end
```

I thought of refactoring `ExclusionRules` to a similarly defined `exclude_example?`, but left it if the later refactoring. Especially since `!exclude_example?` is hard to wrap your head around quickly when called on an instance of `ExclusionRules` :smile:

Reply:
I'm having a think about wether include should take this into account, this just seems like a trap currently.

Reply:
Do you think there could be several rules in the exclusion filter, and we shouldn't check against them with `all?`?

Reply:
I feel like this should be encapsulated somewhere, but for now I think this is fine